Automattic/jetpack
# Recommendations: update button in My Plan to go to Recommendations

#### Changes proposed in this Pull Request:

In #12277 and #12429, we added a "checklist" component to the Jetpack dashboard, as well as a CTA that would lead you to the checklist in Calypso.
We never really expanded on that checklist and its display in the React dashboard, but we now have a Recommendations component we use in Jetpack, since #18437.

This commit removes the elements from the checklist that we never used, and updates the CTA link in "My Plan" to point to the new Recommendations, when those recommendations must be shown to site owners.

#### Jetpack product discussion

* p1HpG7-asW-p2
* p1HpG7-bcl-p2#comment-44627

#### Does this pull request change what data or activity we track or use?

This PR introduces a new Tracks event that will be triggered when clicking on the button. It respects the same rules as the existing Tracks events on that page.

#### Testing instructions:

I would recommend testing this with an Atomic site as well as a Jetpack site.

* Go to Jetpack > Dashboard > My Plan on a site that's connected to WordPress.com
    * If the site is Atomic, you should not see any button
 
<img width="969" alt="Screenshot 2021-02-16 at 17 09 53" src="https://user-images.githubusercontent.com/426388/108090594-13c68e00-707b-11eb-8afc-ae3c33327be8.png">

    * If the site is a Jetpack site, you should see this button

<img width="857" alt="Screenshot 2021-02-16 at 17 09 34" src="https://user-images.githubusercontent.com/426388/108090635-204ae680-707b-11eb-8cb4-570191e9f154.png">

* Clicking on the button should lead you to the Recommendations tab.
* Nothing else should be broken in the dashboard.

#### Proposed changelog entry for your changes:

* N/A
